src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/7.diff.sql

```diff
@@ -0,0 +1,1123 @@
+        AND     PartitionKey = @partitionKey
+    END
+
+    COMMIT TRANSACTION
+END
+GO
+
+/***************************************************************************************/
+-- STORED PROCEDURE
+--    Index instance V7
+--
+-- DESCRIPTION
+--    Adds or updates the various extended query tag indices for a given DICOM instance.
+--
+-- PARAMETERS
+--     @watermark
+--         * The Dicom instance watermark.
+--     @stringExtendedQueryTags
+--         * String extended query tag data
+--     @longExtendedQueryTags
+--         * Long extended query tag data
+--     @doubleExtendedQueryTags
+--         * Double extended query tag data
+--     @dateTimeExtendedQueryTags
+--         * DateTime extended query tag data
+--     @personNameExtendedQueryTags
+--         * PersonName extended query tag data
+-- RETURN VALUE
+--     None
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.IndexInstanceV7
+    @watermark                                                                   BIGINT,
+    @stringExtendedQueryTags dbo.InsertStringExtendedQueryTagTableType_1         READONLY,
+    @longExtendedQueryTags dbo.InsertLongExtendedQueryTagTableType_1             READONLY,
+    @doubleExtendedQueryTags dbo.InsertDoubleExtendedQueryTagTableType_1         READONLY,
+    @dateTimeExtendedQueryTags dbo.InsertDateTimeExtendedQueryTagTableType_2     READONLY,
+    @personNameExtendedQueryTags dbo.InsertPersonNameExtendedQueryTagTableType_1 READONLY
+AS
+BEGIN
+    SET NOCOUNT    ON
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+        DECLARE @partitionKey BIGINT
+        DECLARE @studyKey BIGINT
+        DECLARE @seriesKey BIGINT
+        DECLARE @instanceKey BIGINT
+
+        -- Add lock so that the instance cannot be removed
+        DECLARE @status TINYINT
+        SELECT
+            @partitionKey = PartitionKey,
+            @studyKey = StudyKey,
+            @seriesKey = SeriesKey,
+            @instanceKey = InstanceKey,
+            @status = Status
+        FROM dbo.Instance WITH (HOLDLOCK)
+        WHERE Watermark = @watermark
+
+        IF @@ROWCOUNT = 0
+            THROW 50404, 'Instance does not exists', 1
+        IF @status <> 1 -- Created
+            THROW 50409, 'Instance has not yet been stored succssfully', 1
+
+        -- Insert Extended Query Tags
+
+        -- String Key tags
+        BEGIN TRY
+
+            EXEC dbo.IIndexInstanceCoreV7
+                @partitionKey,
+                @studyKey,
+                @seriesKey,
+                @instanceKey,
+                @watermark,
+                @stringExtendedQueryTags,
+                @longExtendedQueryTags,
+                @doubleExtendedQueryTags,
+                @dateTimeExtendedQueryTags,
+                @personNameExtendedQueryTags
+
+        END TRY
+        BEGIN CATCH
+
+            THROW
+
+        END CATCH
+
+    COMMIT TRANSACTION
+END
+GO
+
+COMMIT TRANSACTION
+
+BEGIN TRANSACTION
```

Comment:
Why are some indexes created online and some not? Why not create all indexes online without a transaction?

Reply:
We include everything in a transaction to make it re-runnable. Since we are not checking IF EXISTS and IF NOT EXISTS for each one, we rely on the transaction to rollback

Reply:
Existing indexes are dropped and created ONLINE and the new indexes are just created OFFLINE. I see we follow this pattern before.